tests/virtualization/universal/windows.pm

```diff
@@ -45,7 +49,14 @@ sub run {
     upload_logs "/tmp/$_-systeminfo.txt"                                                                       foreach (keys %virt_autotest::common::imports);
     assert_script_run "ssh $username\@$_ 'systeminfo' | grep '$virt_autotest::common::imports{$_}->{version}'" foreach (keys %virt_autotest::common::imports);
 
-    remove_guest $_ foreach (keys %virt_autotest::common::imports);
+    remove_guests();
+}
+
+
+sub post_fail_hook {
+    my $self = shift;
+    remove_guests();
+    $self->SUPER::post_fail_hook;
 }
```

Comment:
No, please don't do this. If this module fails we won't have VM to debug.

Reply:
Good point. I left the `post_fail_hook` with a note in there, so that we know that we do this on purpose from now on.